### PR TITLE
fix(settings): Move new user modal to dialog + minor refactoring

### PR DESCRIPTION
**Pull request by Pytal**
## Summary

- Move from NcModal to NcDialog to fix some style issues e.g. close button overlapping scrollbar, use built-in `name` heading for consistent styles
- Some small refactors
- Adjust "user" to "account" in dialog, will make a future followup for some of the other places
- Autofocus username input

Before | After
--- | ---
image | image

| |
| --- |
| image |

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by susnux — APPROVED** [review]

**Comment by Pytal** [comment]
Adjusted a few more places that ripgrep didn't find last time 🙈

File: apps/settings/src/components/Users/NewUserDialog.vue [hunk 1]
```diff
@@ -289,18 +280,18 @@ export default {
 				})
 
 				this.$emit('reset')
-				this.$refs.username?.$refs?.inputField?.$refs?.input?.focus?.()
-				this.$emit('close')
+				this.$refs.username?.focus?.()
+				this.$emit('closing')
```
Comment (susnux) [inline apps/settings/src/components/Users/NewUserDialog.vue:284]: why renaming it here?

Comment (Pytal) [reply apps/settings/src/components/Users/NewUserDialog.vue:284]: This is so we can just listen to `closing` from the parent as we already listen to NcDialog `closing` with `v-on="$listeners"`

Comment (susnux) [reply apps/settings/src/components/Users/NewUserDialog.vue:284]: Ah I see, a bit inconsistent naming on the vue library side (everywhere else "close" is emitted on pressing the button).

**Review by skjnldsv — APPROVED** [review]

**Comment by skjnldsv** [comment]
`users.cy.ts` is failing :)

**Comment by skjnldsv** [comment]
Maybe fixes this https://github.com/nextcloud/server/issues/41011 ?

**Comment by Pytal** [comment]
> Maybe fixes this #41011 ?

Only the not closing on click outside part, other parts still open :)

**Pull request merged**